src/memray/_memray/record_writer.h

```diff
@@ -59,6 +63,13 @@ class RecordWriter
     HeaderRecord d_header{};
     TrackerStats d_stats{};
     thread_id_t d_lastTid{};
+    uintptr_t d_lastInstructionPointer{};
```

Comment:
A bit of a nit, but I think we should encapsulate all of these into some `state` struct for clarity.